bundle/config/resources/job.go

```diff
@@ -47,3 +47,36 @@ func (j *Job) MergeJobClusters() error {
 	j.JobClusters = output
 	return nil
 }
+
+// MergeJobTasks merges job tasks with the same key.
+// The job tasks field is a slice, and as such, overrides are appended to it.
+// We can identify a job task by its task key, however, so we can use this key
+// to figure out which definitions are actually overrides and merge them.
```

Comment:
s/job tasks/tasks/, we're in scope of the job already, and the field is named "tasks"